marklogic-data-hub-central/ui/e2e/hc-qa-project/build.gradle

```diff
@@ -49,3 +52,31 @@ task loadThesaurus(type: com.marklogic.gradle.task.MlcpTask) {
 }
 
 hubDeployAsDeveloper.finalizedBy("loadThesaurus")
+
+class DocCount extends HubTask {
+    DatabaseClient client
+    String script
+    @TaskAction
+    def getDocCount() {
+        try {
+            int result = Integer.parseInt(client.newServerEval().javascript(script).evalAs(String.class));
+            if(result != 34) {
+                throw new GradleException("Record count did not match. Expected 34 in ${client.getDatabase()} database, got "+ result)
+            } else {
+                println("Flow/steps were run successfully and got expected number of records in ${client.getDatabase()} database")
+            }
+        } finally {
+            client.release()
+        }
+    }
+}
+
+task verifyStagingCounts(type: DocCount) {
+    client = hubConfig.newStagingClient()
+    script = "fn.count(fn.collection(['loadCustomersJSON', 'loadCustomersXML', 'order-input', 'loadPersonJSON']))"
```

Comment:
Quick note - fn.count on fn.collection works fine if the counts are in the thousands or less, maybe tens of thousands. But it forces every document to be read in. 

The faster approach is cts.estimate(cts.collectionQuery(['...'])) . That will return a subsecond response for any number of results.